src/cosmicqc/analyze.py

```diff
@@ -201,6 +211,10 @@ def label_outliers(
                 Full dataframe with optional scores and outlier boolean column.
     """
 
+    # interpret the df as SCDataFrame
```

Comment:
ah! Same thing here :)

Maybe the solution is as simple as adding docstrings to each function definition?

Reply:
I an also envision a modification to the `SCDataFrame` testing what to do if `data` is itself an `SCDataFrame` class. This might also remove the need for the current `isinstance()` based approach.

Reply:
Thanks, that's a great idea and a lot simpler! I'll update accordingly.